File: examples/python-repo/ecommerce/application/payment/payment_dto.py
class PaymentDto:
    def __init__(self, payment_id, order_id, amount, method, status):
        self.id = payment_id
        self.order_id = order_id
        self.amount = amount
        self.method = method
        self.status = status

    @classmethod
    def from_domain(cls, payment):
        return cls(
            payment.id,
            payment.order_id,
            str(payment.amount),
            payment.method,
            payment.status,
        )

File: examples/java-repo/src/main/java/com/example/ecommerce/application/payment/PaymentDto.java
package com.example.ecommerce.application.payment;

import com.example.ecommerce.domain.payment.Payment;

public class PaymentDto {

    public final long id;
    public final long orderId;
    public final String amount;
    public final String method;
    public final String status;

    public PaymentDto(long id, long orderId, String amount, String method, String status) {
        this.id = id;
        this.orderId = orderId;
        this.amount = amount;
        this.method = method;
        this.status = status;
    }

    public static PaymentDto fromDomain(Payment payment) {
        return new PaymentDto(
            payment.getId(),
            payment.getOrderId(),
            payment.getAmount().toString(),
            payment.getMethod(),
            payment.getStatus().name());
    }
}
